traits/observation/observe.py

```diff
@@ -8,23 +8,38 @@
 #
 # Thanks for using Enthought open source!
 
+import asyncio
+
 from traits.observation._observe import add_or_remove_notifiers
 from traits.observation.expression import compile_expr
 
+#: Set to hold references to active async traits handlers.
+_active_handler_tasks = set()
```

Comment:
I'm wondering whether we need this to be thread-local, given that different threads might have different current event loops. It's _probably_ okay to have multiple threads modifying a single `set` simultaneously, if there's no overlap in the actual elements being added / removed. Not sure whether that's still going to be true in free-threaded Python 3.13, though. (OTOH, we've got significant work to do if we ever want to make Traits compatible with free-threading.)

Reply:
I _think_ this is fine - each object in the set is independent and there should never be the same task created twice in the same thread. I can see a problem if somehow the task gets finished before the callback is added, but I don't know if that is possible.

Reply:
> each object in the set is independent and there should never be the same task created twice in different threads

Agreed, but the set structure itself would be shared between calls. Right now I think the GIL protects us, but that may eventually no longer be true.

> I can see a problem if somehow the task gets finished before the callback is added, but I don't know if that is possible.

Yep, I don't think that's possible. The task can't start executing until we hit a yield point, so it won't even start until after the callback is added.

Reply:
> Agreed, but the set structure itself would be shared between calls. Right now I think the GIL protects us, but that may eventually no longer be true.

Not sure exactly what the free-threading folks are proposing, but if threaded code needs to worry about adding locks around shared containers at the Python level, it probably won't get adopted widely.  But if it does there'll be plenty of warning I suspect.